src/GridFieldNestedForm.php

```diff
@@ -0,0 +1,435 @@
+
+    private static $allowed_actions = [
+        'handleNestedItem'
+    ];
+
+    private static $max_nesting_level = 10;
+    
+    /**
+     * @var string
+     */
+    protected $name;
+    protected $expandNested = false;
+    protected $forceCloseNested = false;
+    protected $gridField = null;
+    protected $record = null;
+    protected $relationName = 'Children';
+    protected $inlineEditable = false;
+    protected $canExpandCheck = null;
+    protected $maxNestingLevel = null;
+    
+    public function __construct($name = 'NestedForm')
+    {
+        $this->name = $name;
+    }
+    
+    /**
+     * Get the grid field that this component is attached to
+     * @return GridField
+     */
+    public function getGridField()
+    {
+        return $this->gridField;
+    }
+
+    /**
+     * Get the relation name to use for the nested grid fields
+     * @return string
+     */
+    public function getRelationName()
+    {
+        return $this->relationName;
+    }
+    
+    /**
+     * Set the relation name to use for the nested grid fields
+     * @param string $relationName
+     */
+    public function setRelationName($relationName)
+    {
+        $this->relationName = $relationName;
+        return $this;
+    }
+
+    /**
+     * Get whether the nested grid fields should be inline editable
+     * @return boolean
+     */
+    public function getInlineEditable()
+    {
+        return $this->inlineEditable;
+    }
+
+    /**
+     * Set whether the nested grid fields should be inline editable
+     * @param boolean $editable
+     */
+    public function setInlineEditable($editable)
+    {
+        $this->inlineEditable = $editable;
+        return $this;
+    }
+    
+    /**
+     * Set whether the nested grid fields should be expanded by default
+     * @param boolean $expandNested
+     */
+    public function setExpandNested($expandNested)
+    {
+        $this->expandNested = $expandNested;
+        return $this;
+    }
+
+    /**
+     * Set whether the nested grid fields should be forced closed on load
+     * @param boolean $forceClosed
+     */
+    public function setForceClosedNested($forceClosed)
+    {
+        $this->forceCloseNested = $forceClosed;
+        return $this;
+    }
+
+    /**
+     * Set a callback to check which items in this grid that should show the expand link
+     * for nested gridfields. The callback should return a boolean value.
+     * @param callable $checkFunction
+     */
+    public function setCanExpandCheck($checkFunction)
+    {
+        $this->canExpandCheck = $checkFunction;
+        return $this;
+    }
+
+    /**
+     * Set the maximum nesting level allowed for nested grid fields
+     * @param int $level
+     */
+    public function setMaxNestingLevel($level)
+    {
+        $this->maxNestingLevel = $level;
+        return $this;
+    }
+
+    public function getMaxNestingLevel()
+    {
+        return $this->maxNestingLevel ?: $this->config()->max_nesting_level;
```

Comment:
```suggestion
        return $this->maxNestingLevel ?: static::config()->get('max_nesting_level');
```
Config is a static method so should be accessed statically.
The get method should be explicitly used instead of relying on the magic `__get()`, to make it clearer what's happening.

If there's any other usage of config in this PR, please also update those.